### PR TITLE
[android] Remove unused code

### PR DESCRIPTION
**Pull request by rtsisyk**

### CODE REVIEW

**Comment by rtsisyk** [comment]
```
48462269 OrganicMaps-23092808-fdroid-beta.apk before
45710406 OrganicMaps-23092905-fdroid-beta.apk after
```

Minus 2.62MB (~6%).

File: android/app/src/main/java/app/organicmaps/Map.java
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/Map.java:1]: The methods you removed from this file are used in Android Auto.

File: android/app/src/main/java/app/organicmaps/api/Const.java [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 public class Const
 {
-  // Common
-  static final String API_SCHEME = "om://";
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/api/Const.java:6]: This line will be used in AA

File: android/app/src/main/java/app/organicmaps/Framework.java [hunk 2]
```diff
@@ -181,27 +190,10 @@ public static void setSpeedCamerasMode(@NonNull SettingsPrefsFragment.SpeedCamer
 
   public static native String nativeGetGe0Url(double lat, double lon, double zoomLevel, String name);
 
-  public static native String nativeGetAddress(double lat, double lon);
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/Framework.java:184]: Used in AA

Comment (AndrewShkrob) [reply android/app/src/main/java/app/organicmaps/Framework.java:184]: If this logic is incorrect, I can fix it:
```java
    String address = mMapObject.getAddress();
    if (address.isEmpty())
      address = Framework.nativeGetAddress(mMapObject.getLat(), mMapObject.getLon());
    if (!address.isEmpty())
      builder.addText(address);
```

File: android/app/src/main/java/app/organicmaps/bookmarks/data/MapObject.java [hunk 3]
```diff
@@ -263,27 +248,11 @@ public String getMetadata(Metadata.MetadataType type)
     return res == null ? "" : res;
   }
 
-  public boolean hasMetadata()
-  {
-    return mMetadata != null && !mMetadata.isEmpty();
-  }
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/bookmarks/data/MapObject.java:269]: Used in AA

File: android/app/src/main/java/app/organicmaps/Framework.java (hunk 2 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/Framework.java:184]: Please raise a PR to master?

File: android/app/src/main/java/app/organicmaps/bookmarks/data/MapObject.java (hunk 3 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/bookmarks/data/MapObject.java:269]: I guess how?

File: android/app/src/main/java/app/organicmaps/search/SearchFragment.java [hunk 4]
```diff
@@ -46,8 +45,8 @@
 import java.util.List;
 
 public class SearchFragment extends BaseMwmFragment
-                         implements OnBackPressListener,
-                                    NativeSearchListener,
+                         implements
+    NativeSearchListener,
                                     CategoriesAdapter.CategoriesUiListener
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/search/SearchFragment.java:50]: Fix formatting

File: android/app/src/main/java/app/organicmaps/search/SearchFragment.java [hunk 5]
```diff
@@ -570,6 +563,7 @@ public SearchToolbarController requireController()
   private static class BadStorageCommand extends HiddenCommand.BaseHiddenCommand
   {
     @NonNull
+    final
     Context mContext;
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/search/SearchFragment.java:567]: Fix formatting

File: android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java [hunk 6]
```diff
@@ -59,18 +57,11 @@ public void run()
   @NonNull
   private Context mContext;
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java:58]: final

File: android/app/src/main/java/app/organicmaps/bookmarks/data/MapObject.java (hunk 3 above)
Comment (AndrewShkrob) [reply android/app/src/main/java/app/organicmaps/bookmarks/data/MapObject.java:269]:
```java
@Nullable
  public static Row getPlaceOpeningHoursRow(@NonNull MapObject place, @NonNull CarContext context)
  {
    if (!place.hasMetadata())
      return null;
    // ...
```

```java
@NonNull
  private static String getFirstPhoneNumber(@NonNull MapObject mapObject)
  {
    if (!mapObject.hasMetadata())
      return "";

    final String phones = mapObject.getMetadata(Metadata.MetadataType.FMD_PHONE_NUMBER);
    return phones.split(";", 1)[0];
  }
```

File: android/app/src/main/java/app/organicmaps/Framework.java (hunk 2 above)
Comment (AndrewShkrob) [reply android/app/src/main/java/app/organicmaps/Framework.java:184]: > Please raise a PR to master?

?

**Comment by biodranik** [comment]
Are there any non-risky removals that can be safely merged without detailed review in a separate PR?

**Comment by biodranik** [comment]
Linter check is failing btw.

**Comment by rtsisyk** [comment]
> Are there any non-risky removals that can be safely merged without detailed review in a separate PR?

Extracted the first non-risk part https://github.com/organicmaps/organicmaps/pull/6162, PTAL.

**Comment by rtsisyk** [comment]
Extracted https://github.com/organicmaps/organicmaps/pull/6199

**Comment by rtsisyk** [comment]
This PR should be rebased on the latest master before it can be reviewed. The most changes are already in the master.

**Comment by rtsisyk** [comment]
Removal of unused code only. Rebased on the latest master. Please review.

**Comment by Jean-BaptisteC** [comment]
Any idea how many we save spaces ?

**Comment by rtsisyk** [comment]
> Any idea how many we save spaces ?

Not so much, all space was saved by R8 and removal of unused resources. The code doesn't save a lot.

**Review by biodranik — APPROVED** [review]

**Comment by biodranik** [comment]
LGTM

**Review by vng — APPROVED** [review]

**Pull request merged**